packages/client/components/PokerEstimateHeaderCard.tsx

```diff
@@ -4,12 +4,58 @@ import {createFragmentContainer} from 'react-relay'
 import {PokerEstimateHeaderCard_stage} from '../__generated__/PokerEstimateHeaderCard_stage.graphql'
 import PokerEstimateHeaderCardContent from './PokerEstimateHeaderCardContent'
 import PokerEstimateHeaderCardError from './PokerEstimateHeaderCardError'
-import PokerEstimateHeaderCardGitHub from './PokerEstimateHeaderCardGitHub'
 import PokerEstimateHeaderCardParabol from './PokerEstimateHeaderCardParabol'
 
 interface Props {
   stage: PokerEstimateHeaderCard_stage
 }
+
+type Integration = NonNullable<NonNullable<PokerEstimateHeaderCard_stage['task']>['integration']>
+export type IntegrationHeaderFields = {
+  cardTitle: string
+  descriptionHTML: string
+  url: string
+  linkTitle: string
+  linkText: string
+}
+
+const getHeaderFields = (integration: Integration | null): IntegrationHeaderFields | null => {
+  if (!integration) return null
+  const {__typename} = integration
+  switch (__typename) {
+    case 'JiraServerIssue':
```

Comment:
I think we would need to do something in the future to not have a lot of mappings like this. Created a separate  discussion issue: #6374